addons/mail/controllers/discuss.py

```diff
@@ -192,7 +192,9 @@ def fetch_image(self, channel_id, attachment_id, width=0, height=0, **kwargs):
     @http.route('/mail/init_messaging', methods=['POST'], type='json', auth='public')
     def mail_init_messaging(self, **kwargs):
         if not request.env.user.sudo()._is_public():
-            return request.env.user.sudo(False)._init_messaging()
+            if request.env.user._is_internal():
+                return request.env.user.sudo(False)._init_messaging()
+            return request.env.user._init_messaging()
```

Comment:
This change should be done in master directly, and have a test to cover it.

Also I think this change might be too naive and leak potentially unwanted information to portal users.

Reply:
Not master actually, but v15? v16? whatever is the first version where it fails.